modules/browsiRtdProvider.js

```diff
@@ -103,12 +139,97 @@ export function collectData() {
       sw: (win.screen && win.screen.width) || -1,
       sh: (win.screen && win.screen.height) || -1,
       url: `${doc.location.protocol}//${doc.location.host}${doc.location.pathname}`,
+      eu: isEngagingUser(),
+      t: TIMESTAMP,
+      pvid: PVID
     },
-    ...(browsiData ? {us: browsiData} : {us: '{}'}),
-    ...(document.referrer ? {r: document.referrer} : {}),
-    ...(document.title ? {at: document.title} : {})
+    ...(brtd ? { us: brtd } : { us: '{}' }),
+    ...(document.referrer ? { r: document.referrer } : {}),
+    ...(document.title ? { at: document.title } : {}),
+    ...(highestBidMetrics?.rahb ? { rahb: highestBidMetrics.rahb.avg.toFixed(3) } : {}),
+    ...(highestBidMetrics?.uahb ? { uahb: highestBidMetrics.uahb.avg.toFixed(3) } : {}),
+    ...(highestBidMetrics?.lahb ? { lahb: highestBidMetrics.lahb.avg.toFixed(3), lbsa: highestBidMetrics.lahb.age.toFixed(3) } : {}),
   };
-  getPredictionsFromServer(`//${_moduleParams.url}/prebid?${toUrlParams(predictorData)}`);
+  getPredictionsFromServer(`//${_moduleParams.url}/prebid?${toUrlParams(predictorData)}`); // TODO change route
+}
+
+function convertBusData(bus) {
+  try {
+    if (!bus) return undefined;
+    const busObj = JSON.parse(bus);
+    return busObj && Object.keys(busObj).length ? busObj : undefined;
+  } catch (e) {
+    return undefined;
+  }
+}
+
+export function getHighestBidMetrics(bus) {
+  try {
+    const rahbByTimestamp = bus.rahb && Object.keys(bus.rahb).length && getRecentAvgBidByTimestamp(bus.rahb);
```

Comment:
seems like to much cheks, espicaily if you're passing it to function
the function already check for keys, it can also check for undefined